rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -208,8 +735,8 @@ impl StatisticsCollector {
             let max_value = Arc::new(builder.max_value.finish());
             let struct_fields = vec![
                 ArrowField::new("null_count", DataType::Int64, false),
-                ArrowField::new("min_value", field.data_type(), field.nullable),
-                ArrowField::new("max_value", field.data_type(), field.nullable),
+                ArrowField::new("min_value", field.data_type(), true),
+                ArrowField::new("max_value", field.data_type(), true),
```

Comment:
Field should only be nullable for non-primitives, right?

Reply:
Indeed. Changed the collector to use the field's nullability.